src/util.ts

```diff
@@ -351,6 +351,20 @@ export function parseVersion(str: string): Version {
     };
 }
 
+export function tryParseVersion(str: string): Version | undefined {
```

Comment:
To avoid multiple copies of the same code, I would prefer if this was written to call the `parseVersion` function instead.
e.g.
```ts
// warning: code written in browser, not tested
export function tryParseVersion(str: string): Version | undefined {
    try {
        return parseVersion(str);
    } catch {
        return undefined;
    }
}
```